docs/docs/why-gatsby-uses-graphql.md

```diff
@@ -0,0 +1,360 @@
+    "description": "<p>Get your feet into these spaced-out black socks with a Gatsby purple border and heel.</p>",
+    "price": "$10.00",
+    "image": "/images/erin-fox-and-sullivan.jpg"
+  },
+  {
+    "title": "This Purple Hat Is Blazing Fast",
+    "slug": "purple-hat",
+    "description": "<p>Add more blazingly blazing speed to your wardrobe with this solid purple laundered chino twill hat.</p>",
+    "price": "$10.00",
+    "image": "/images/david-bailey-cat-hat.jpg"
+  }
+]
+```
+
+The images need to be added to the `/static/images/` folder. (This is where things start to get hard to manage — the JSON and the images aren’t in the same place.)
+
+Once the JSON and the images are added, you can create product pages by importing the JSON into `gatsby-node.js` and loop through the entries to create pages:
+
+```js:title=gatsby-node.js
+exports.createPages = ({ actions: { createPage } }) => {
+  const products = require("./data/products.json")
+  products.forEach(product => {
+    createPage({
+      path: `/product/${product.slug}/`,
+      component: require.resolve("./src/templates/product.js"),
+      context: {
+        title: product.title,
+        description: product.description,
+        image: product.image,
+        price: product.price,
+      },
+    })
+  })
+}
+```
+
+The product template still uses `pageContext` to display the product data:
+
+```jsx:title=src/templates/product.js
+import React from "react"
+
+const Product = ({ pageContext }) => (
+  <div>
+    <h1>{pageContext.title}</h1>
+    <img
+      src={pageContext.image}
+      alt={pageContext.title}
+      style={{ float: "left", marginRight: "1rem", width: 150 }}
+    />
+    <p>{pageContext.price}</p>
+    <div dangerouslySetInnerHTML={{ __html: pageContext.description }} />
+  </div>
+)
+
+export default Product
+```
+
+Run `gatsby develop`, then open `localhost:8000/product/space-socks/` to see one of the generated products.
+
+![Screenshot of a rendered product page.](./images/why-gql-product-json.png)
+
+This gets the job done, but it has a few shortcomings that are going to get more complicated as time goes on:
+
+1. The images and the product data are in different places in the source code.
+2. The image paths are absolute from the _built_ site, not the source code, which makes it confusing to know how to find them from the JSON.
+3. The images are unoptimized, and any optimization you do would have to be manual.
+4. To create a preview listing of all products, we’d need to pass _all_ of the product info in `context`, which will get unweildy as the number of products increases.
+5. It’s not very obvious where data is coming from in the templates that render the pages, so updating the data might be confusing later.
+
+To overcome these limitations, Gatsby introduces GraphQL as a data management layer.
+
+## Create pages using GraphQL
+
+There’s a bit more up-front setup required to get data into GraphQL, but the benefits often far outweigh the cost.
```

Comment:
Maybe a bit stronger here? We can/should strongly encourage GraphQL.

```suggestion
There’s a bit more up-front setup required to get data into GraphQL, but the benefits far outweigh the cost.
```

Reply:
Yeah, I could go either way. I think there are definitely simple use cases where GraphQL isn’t needed, and we shouldn’t necessarily push folks toward GraphQL if they don't _need_ it. That being said, we can (and should) have a point of view, so I'm happy to make this a bit more strongly worded.